subsys/bluetooth/audio/ascs.c

```diff
@@ -1331,6 +1331,49 @@ struct ascs_parse_result {
 	size_t count;
 };
 
+static bool ascs_valid_metadata_type(uint8_t type, uint8_t len)
+{
+	switch (type) {
+	case BT_AUDIO_METADATA_TYPE_PREF_CONTEXT:
+	case BT_AUDIO_METADATA_TYPE_STREAM_CONTEXT:
+		if (len != 2) {
+			return false;
+		}
+
+		return true;
+	case BT_AUDIO_METADATA_TYPE_STREAM_LANG:
+		if (len != 3) {
+			return false;
+		}
+
+		return true;
+	case BT_AUDIO_METADATA_TYPE_PARENTAL_RATING:
+		if (len != 1) {
+			return false;
+		}
+
+		return true;
+	case BT_AUDIO_METADATA_TYPE_EXTENDED: /* 1 - 255 octets */
+	case BT_AUDIO_METADATA_TYPE_VENDOR: /* 1 - 255 octets */
+		if (len < 1) {
+			return false;
+		}
+
+		return true;
+	case BT_AUDIO_METADATA_TYPE_CCID_LIST: /* 2 - 254 octets */
+		if (len < 2) {
+			return false;
+		}
+
+		return true;
+	case BT_AUDIO_METADATA_TYPE_PROGRAM_INFO: /* 0 - 255 octets */
+	case BT_AUDIO_METADATA_TYPE_PROGRAM_INFO_URI: /* 0 - 255 octets */
+		return true;
+	default:
+		return false;
+	}
+}
```

Comment:
So you will not allow to establish stream it there's unknown metadata type? What if there will be new type added?

Reply:
Yes :) 

Test case `ASCS/SR/SPE/BI-09-C [Update Metadata – Invalid Parameters]` specifies that we shall reject the metadata operation with LTV with an RFU value. 

It isn't specifically written by the Generic Audio assigned numbers that the non-allocated types are RFU, so I just assumed that they were. @asbjornsabo Input here?

Reply:
Well, I think they must be RFU in the sense that only BT SIG will be allowed to add new types.
But the SIG are adding new types in some instances (in periodic advertising for broadcast), so it seems reasonable to allow for unknown type.  OTOH - the test you are referring to seems very clear.
Suggest to add a TODO until we have clarified.

Reply:
@asbjornsabo This is common behavior in Bluetooth; I don't think we need to clarify it with anything but us. 

To pass the test, either we or the application needs to check for the metadata type. I am fine with letting the application do it, but at the same time, since Zephyr is not a library, adding support for more metadata types is just as easy to add to Zephyr as it is to add to the application IMO. 

This discussion is unfortunately taking place in two comment threads (see also my comment here: https://github.com/zephyrproject-rtos/zephyr/pull/44522#discussion_r896618448). 

In the end, as @Casper-Bonde-Bose mentioned, metadata is entirely informative at this layer, and as such I agree that we can just offload that to a higher layer (will make that change).